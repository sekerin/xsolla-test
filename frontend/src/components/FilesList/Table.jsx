import React, { Component } from 'react';
import PropTypes            from 'prop-types';
import { Table }            from 'react-bootstrap';
import TableItem            from './TableItem';

const propsTypes = {
  data: PropTypes.any
};

class TableModule extends Component {
  constructor() {
    super();

    this.getData = this.getData.bind(this);
  }

  getData() {
    const data = [];

    this.props.data.forEach((item) => {
      data.push(
        <TableItem
          key={item.name}
          name={item.name}
        />
      );
    });
    return (data) ? data : 'No objects';
  }

  render() {
    return (
      <Table
        striped
        bordered
        condensed
        hover
        responsive
      >
        <thead>
          <tr>
            <td>Name</td>
            <td>Actions</td>
          </tr>
        </thead>
        <tbody>{this.getData()}</tbody>
      </Table>
    );
  }
}

TableModule.propTypes = propsTypes;

export default TableModule;
