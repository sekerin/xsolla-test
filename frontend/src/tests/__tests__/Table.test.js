import React from 'react';

import { shallow }       from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Table from '../../components/FilesList/Table';

describe('Test Table component', () => {
  it('Empty Table', () => {
    const wrapper = shallow(<Table onDelete={{}} data={[]}/>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('Table with data', () => {
    const wrapper = shallow(<Table onDelete={{}} data={[{ name: 'file' }]}/>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
