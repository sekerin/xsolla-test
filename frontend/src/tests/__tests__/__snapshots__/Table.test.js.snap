// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Table component Empty Table 1`] = `
<Table
  bordered={true}
  bsClass="table"
  condensed={true}
  hover={true}
  responsive={true}
  striped={true}
>
  <thead>
    <tr>
      <td>
        Name
      </td>
      <td>
        Actions
      </td>
    </tr>
  </thead>
  <tbody />
</Table>
`;

exports[`Test Table component Table with data 1`] = `
<Table
  bordered={true}
  bsClass="table"
  condensed={true}
  hover={true}
  responsive={true}
  striped={true}
>
  <thead>
    <tr>
      <td>
        Name
      </td>
      <td>
        Actions
      </td>
    </tr>
  </thead>
  <tbody>
    <TableItem
      key="file"
      name="file"
      onDelete={[MockFunction]}
      onEditItem={[MockFunction]}
    />
  </tbody>
</Table>
`;
